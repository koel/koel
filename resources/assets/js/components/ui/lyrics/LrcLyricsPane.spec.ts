import { describe, expect, it } from 'vitest'
import { createHarness } from '@/__tests__/TestHarness'
import { screen } from '@testing-library/vue'
import Component from './LrcLyricsPane.vue'

describe('lrcLyricsPane.vue', () => {
  const h = createHarness()

  const renderComponent = (lyrics?: Array<{ time: number, text: string }>) => {
    lyrics = lyrics ?? [
      { time: 10.5, text: 'First line' },
      { time: 15.2, text: 'Second line' },
      { time: 20.0, text: 'Third line' },
    ]

    return h.render(Component, {
      props: {
        lyrics,
        fontSize: '1rem',
      },
      global: {
        stubs: {
          LrcLyricsLine: h.stub('lyrics-line'),
        },
      },
    })
  }

  it('renders all lyrics lines', () => {
    renderComponent()
    expect(screen.queryAllByTestId('lyrics-line')).toHaveLength(3)
  })

  it('renders empty when no lyrics provided', () => {
    renderComponent([])
    expect(screen.queryAllByTestId('lyrics-line')).toHaveLength(0)
  })
})
