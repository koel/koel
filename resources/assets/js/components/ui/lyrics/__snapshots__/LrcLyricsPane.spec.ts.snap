// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`lrcLyricsPane.vue > renders 1`] = `
<article data-v-6db450e8="" class="overflow-y-auto space-y-2">
  <p data-v-95602d25="" data-v-6db450e8="" class="relative leading-relaxed origin-left transition-all duration-200 cursor-default text-k-fg hover:!opacity-100" style="opacity: 0.75;">First line</p>
  <p data-v-95602d25="" data-v-6db450e8="" class="relative leading-relaxed origin-left transition-all duration-200 cursor-default text-k-fg hover:!opacity-100" style="opacity: 0.5;">Second line</p>
  <p data-v-95602d25="" data-v-6db450e8="" class="relative leading-relaxed origin-left transition-all duration-200 cursor-default text-k-fg hover:!opacity-100" style="opacity: 0.25;">Third line</p>
</article>
`;
